src/main/scanLogic/scanRunners/binaryRunner.ts

```diff
@@ -92,6 +126,71 @@ export abstract class BinaryRunner {
         await Promise.all(tasksInfo.tasks);
     }
 
+    /**
+     * Create the needed environment variables for the runner to run
+     * @returns list of environment variables to use while executing the runner or unidentified if credential not set
+     */
+    private createEnvForRun(): NodeJS.ProcessEnv | undefined {
+        if (this._connectionManager.areXrayCredentialsSet()) {
+            // Platform information
+            let binaryVars: NodeJS.ProcessEnv = {
+                JF_PLATFORM_URL: this._connectionManager.url
+            };
+            if (this._connectionManager.accessToken) {
+                binaryVars.JF_TOKEN = this._connectionManager.accessToken;
+            } else {
+                binaryVars.JF_USER = this._connectionManager.username;
+                binaryVars.JF_PASS = this._connectionManager.password;
+            }
+            let proxyHttpUrl: string | undefined;
+            let proxyHttpsUrl: string | undefined;
+            // Proxy information - environment variable
+            let proxy_env_http: string | undefined = process.env['HTTP_PROXY'];
+            if (proxy_env_http) {
+                proxyHttpUrl = proxy_env_http;
+            }
+            let proxy_env_https: string | undefined = process.env['HTTPS_PROXY'];
+            if (proxy_env_https) {
+                proxyHttpsUrl = proxy_env_https;
+            }
+            // Proxy information - vscode configuration override
+            let optional: IProxyConfig | boolean = ConnectionUtils.getProxyConfig();
+            if (optional) {
+                let proxyConfig: IProxyConfig = <IProxyConfig>optional;
+                let proxyUrl: string = proxyConfig.host + ':' + proxyConfig.port;
+                proxyHttpUrl = proxyUrl;
+                proxyHttpsUrl = proxyUrl;
+            }
+            // Proxy auth
+            let authOptional: string | undefined = Configuration.getProxyAuth();
+            if (authOptional) {
+                let authDecoded: string = Buffer.from(
+                    authOptional.startsWith('Basic ') ? authOptional.substring('Basic '.length) : authOptional,
+                    'base64'
+                ).toString('binary');
+                // We expect the decoded auth string to be in the format: <proxy-user>:<proxy-password>
+                if (proxyHttpUrl) {
+                    proxyHttpUrl = authDecoded + '@' + proxyHttpUrl;
+                }
+                if (proxyHttpsUrl) {
+                    proxyHttpsUrl = authDecoded + '@' + proxyHttpsUrl;
+                }
```

Comment:
This will not work for bearer tokens. For them you can do the following:
```typescript
if (proxyHttpUrl) {
    proxyHttpUrl = proxyHttpUrl + "?access_token=<the-access-token>";
}
if (proxyHttpsUrl) {
    proxyHttpUrl = proxyHttpsUrl + "?access_token=<the-access-token>";
}
```

See https://stackoverflow.com/questions/59356703/api-passing-bearer-token-to-get-http-url